### PR TITLE
refactor: move linked roles to core package

### PR DESCRIPTION
**Pull request by apteryxxyz**
Part of #99

### CODE REVIEW

File: website/tsconfig.json [hunk 1]
```diff
@@ -29,7 +29,8 @@
 		"**/*.ts",
 		"**/*.tsx",
 		".next/types/**/*.ts",
-		"postcss.config.cjs"
+		"postcss.config.cjs",
+		"content/carbon/plugins/index.mdx"
```
Comment (thewilloftheshadow) [inline website/tsconfig.json:33]: extra tsconfig entry?

Comment (apteryxxyz) [reply website/tsconfig.json:33]: ah another instance of the editor updating imports and such when i dont want it to

Comment (thewilloftheshadow) [reply website/tsconfig.json]: figured as much 😂

**Review by thewilloftheshadow — CHANGES_REQUESTED** [review]
The code looks good to me, but can we tweak the api generation for the docs
CleanShot 2024-09-22 at 10 03 45@2x

**Review by thewilloftheshadow — APPROVED** [review]

**Pull request merged**